src/libs/ajax/data-table-providers/WdsDataTableProvider.ts

```diff
@@ -90,48 +91,35 @@ const getRelationParts = (val: unknown): string[] => {
   // WDS looks for Kubernetes deployment statuses (such as RUNNING or PROVISIONING), expressed by Leo
   // See here for specific enumerations -- https://github.com/DataBiosphere/leonardo/blob/develop/core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/kubernetesModels.scala
   // look explicitly for a RUNNING app named 'wds-${app.workspaceId}' -- if WDS is healthy and running, there should only be one app RUNNING
   // an app may be in the 'PROVISIONING', 'STOPPED', 'STOPPING', which can still be deemed as an OK state for WDS
   const healthyStates = ['RUNNING', 'PROVISIONING', 'STOPPED', 'STOPPING'];
 
-  // WDS appType is checked first and takes precedence over CROMWELL apps in the workspace
-  const wdsAppTypes = ['WDS', 'CROMWELL'];
-  for (const wdsAppType of wdsAppTypes) {
-    const namedApp = apps.filter(
-      (app) =>
-        app.appType === wdsAppType && app.appName === `wds-${app.workspaceId}` && healthyStates.includes(app.status)
-    );
-    if (namedApp.length === 1) {
-      return namedApp[0];
-    }
-
-    // Failed to find an app with the proper name, look for a RUNNING WDS or CROMWELL app
-    const runningWdsApps = apps.filter((app) => app.appType === wdsAppType && app.status === 'RUNNING');
-    if (runningWdsApps.length > 0) {
-      // Evaluate the earliest-created WDS app
-      runningWdsApps.sort(
-        (a, b) => new Date(a.auditInfo.createdDate).valueOf() - new Date(b.auditInfo.createdDate).valueOf()
-      );
-      return runningWdsApps[0];
-    }
-
-    // If we reach this logic, we have more than one Leo app with the associated workspace Id...
-    const allWdsApps = apps.filter(
-      (app) => app.appType === wdsAppType && ['PROVISIONING', 'STOPPED', 'STOPPING'].includes(app.status)
-    );
-    if (allWdsApps.length > 0) {
-      // Evaluate the earliest-created WDS app
-      allWdsApps.sort(
-        (a, b) => new Date(a.auditInfo.createdDate).valueOf() - new Date(b.auditInfo.createdDate).valueOf()
-      );
-      return allWdsApps[0];
-    }
-  }
-
-  return '';
+  // @ts-expect-error Leo type definitions do not include WDS
+  const eligibleApps = apps.filter((app) => app.appType === 'WDS' || app.appType === 'CROMWELL');
+
+  const prioritizedApps = _.sortBy(
+    [
+      // Prefer WDS app with the proper name
+      // @ts-expect-error Leo type definitions do not include WDS
+      (app) => (app.appType === 'WDS' && app.appName === `wds-${app.workspaceId}` ? -1 : 1),
+      // Prefer WDS apps over CROMWELL apps
+      // @ts-expect-error Leo type definitions do not include WDS
+      (app) => (app.appType === 'WDS' ? -1 : 1),
+      // Prefer running apps
+      (app) => (app.status === 'RUNNING' ? -1 : 1),
+      // Prefer healthy apps
+      (app) => (healthyStates.includes(app.status) ? -1 : 1),
+      // Prefer apps created earlier
+      (app) => new Date(app.auditInfo?.createdDate).valueOf(),
+    ],
+    eligibleApps
+  );
```

Comment:
I don't think this logic is correct, just from reading the code; I did not try to write unit tests or test it empirically.

This should never, under any circumstance, return an app that is anything other than `appType === 'WDS'`. It looks like this logic allows that, in a corner case.

Reply:
The current implementation returns apps with `appType === 'CROMWELL'`, though it prefers WDS.

This can be seen in the unit tests for `resolveWdsUrl`.
https://github.com/DataBiosphere/terra-ui/blob/35ddbc59e4e6f5c587ca45a34e0a734bf3fb80f3/src/libs/ajax/data-table-providers/WdsDataTableProvider.test.ts#L770-L892

Reply:
Ah - I think I understand what's going on here, this is support for the much older `CROMWELL` apps where WDS and CBAS were still bundled. I suppose we leave it in for now, but at some point soon we should remove that logic and stop supporting super-old deployments.